web/client/components/mapcontrols/measure/__tests__/MeasureDialog-test.jsx

```diff
@@ -127,5 +127,23 @@ describe("test the MeasureDialog", () => {
         expect(spyInit.calls.length).toBe(1);
         expect(spyInit).toHaveBeenCalledWith(defaultOptions);
     });
+    it('should trigger onInit if defaultOptions are updated', () => {
+        const handler = {
+            onInit: () => {}
+        };
+        const spyInit = expect.spyOn(handler, "onInit");
+        const measurement = {};
+        let defaultOptions = {showAddAsAnnotation: true};
+        ReactDOM.render(
+            <MeasureDialog
+                show
+                onInit={handler.onInit}
+                defaultOptions={defaultOptions}
+                measurement={measurement}/>, document.getElementById("container"));
+        setTimeout(() => {
```

Comment:
This test doesn't really check the fix. Please avoid async tests without done callback and, in order to check property update, you have to re-render the component.